### PR TITLE
Delete `solid_body_rotation_split.py` demo

### LINKED ISSUE #171: Delete demo on Solid body rotation with multiple prognostic variables
I think that the `solid_body_rotation_split.py` demo is written in a non-goalie way, which became apparent when working on #169. For example, the way it's written would not allow us to extend it to a goal-oriented approach.

It's also a very very specific case where all PDEs are the same. So I'm not sure how useful it is in general, and may even be misleading/confusing. E.g. I think that adding the `field` kwarg to `get_form` may be quite confusing to new users.

I would either:
- rewrite it in a goalie way - i.e. without the shortcut for different fields
- or remove it completely, since we have `gray_scott_split.py` to demonstrate how to deal with multiple variables (which I think is the point of the `solid_body_rotation_split.py` demo

What do you think @jwallwork23?

### PR DESCRIPTION
**Pull request by ddundo**
Closes #171.

Also updated `solid_body_rotation.py` to no longer account for the split case.

### CODE REVIEW

File: demos/solid_body_rotation.py [hunk 1]
```diff
@@ -271,10 +269,6 @@ def qoi():
             for sol in sols["adjoint"][i]:
                 adj_outfile.write(sol)
 
-# In the `next demo <./solid_body_rotation_split.py.html>`__,
-# we consider solving the same problem, but splitting the solution
-# field into multiple components.
-#
```
Comment (jwallwork23) [inline demos/solid_body_rotation.py:277]: Please provide a link to the first Gray-Scott demo, which is now the next one in the sequence.

**Review by jwallwork23 — CHANGES_REQUESTED** [review]
Thanks @ddundo. Please address my comment and also open a PR to update https://github.com/mesh-adaptation/mesh-adaptation-docs accordingly.

File: demos/solid_body_rotation.py (hunk 1 above)
Comment (ddundo) [reply demos/solid_body_rotation.py]: Fixed in 09f4999e0ec28a5ad279975143eb80ace3601f97. I cancelled the workflow since it's just a comment.

**Comment by ddundo** [comment]
Thanks @jwallwork23! I addressed your comment and opened https://github.com/mesh-adaptation/mesh-adaptation-docs/issues/18.

**Review by jwallwork23 — APPROVED** [review]
Thanks @ddundo.

**Pull request merged**